File: String/Palindrome/Palindrome.cc
class Palindrome {
	public:
		bool isPalindrome(string &s) {
			if(s.size() == 0)
				return true;
			
			int l = 0;
			int r = s.size() - 1;
			
			while(l < r) {
				if(isalnum(s[l])) {				
					l += 1;
					continue;
				}
				if(isalnum(s[r])) {
					r -= 1;
					continue;
				}
				
				if(isalnum(s[l]) == isalnum(s[r])) {
					--l;
					++r;					
				} else 
					return false;
					
			}
			return true;
		}
};

File: String/Palindrome/Palindrome.java
public class Palindrome {
	public boolean isPalindrome(String s) {
		if(s == null || s.empty())
			return true;
		
		int l = 0;
		int r = s.length() - 1;
		
		while(l < r) {
			if(!Character.isLetterOrNum(s.CharAt[l])) {
				l++;
				continue;
			}
			
			if(!Character.isLetterOrNum(s.CharAt[r])) {
				r--;
				continue;
			}
			
			if(Character.isLetterOrNum(s.CharAt[l]) == Character.isLetterOrNum(s.CharAt[r])) {
				l++;
				r--;
			} else 
				return false;
		}
		return true;
	}
}